.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,106 @@
+name: Run snarkVM Benchmarks
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - 'testnet3'
+      - 'testnet3.3'
+  push:
+    branches:
+      - 'testnet3'
+      - 'testnet3.3'
+
+jobs:
+  # Run benchmarks and stores the output to a file
+  benchmark:
+    name: Benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-bench-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Benchmark algorithms
+        run: |
+          cd algorithms
+          cargo bench --bench variable_base -- --output-format bencher | tee -a ../output.txt
+          cargo bench --bench poseidon_sponge -- --output-format bencher | tee -a ../output.txt
+          cargo bench --bench marlin -- --output-format bencher | tee -a ../output.txt
```

Comment:
The FFT benches are quite important to not regress on, but I understand the running time is almost a non-starter for CI. However, the following approaches might work

1. *Create a reduced set of benches for CI wherein we sample a little bit less on each CI pull*: Our benches are quite exhaustive. For instance the merkle tree bench is O(number_of_starting_leaves*added_number_of_leaves) and our FFT benches bench  BENCHMARK_MIN_DEGREE..BENCHMARK_MAX_DEGREE. However instead of being `exhaustively` confident in the benches we could get `statistical confidence` by maybe just running 2 degrees within that range and a similar approach for the merkle tree & marlin benches.

2. Add the ability to run bench CI on a manual trigger and require the benchmark is run before an actual merge. This could cut the number of CI runs a lot.